File: polocloud-suite/src/main/java/dev/httpmarco/polocloud/suite/commands/type/KeywordArgument.java
package dev.httpmarco.polocloud.suite.commands.type;

import dev.httpmarco.polocloud.suite.commands.CommandArgument;
import dev.httpmarco.polocloud.suite.commands.CommandContext;

import java.util.List;

public final class KeywordArgument extends CommandArgument<String> {

    public KeywordArgument(String key) {
        super(key);
    }

    @Override
    public List<String> defaultArgs(CommandContext context) {
        return List.of(key());
    }

    @Override
    public String wrongReason() {
        return "";
    }

    @Override
    public String buildResult(String input) {
        return "";
    }

    @Override
    public boolean predication(String rawInput) {
        return super.predication(rawInput) && rawInput.equalsIgnoreCase(key());
    }
}

File: agent/src/main/java/dev/httpmarco/polocloud/agent/runtime/local/terminal/commands/type/KeywordArgument.kt
package dev.httpmarco.polocloud.agent.runtime.local.terminal.commands.type

import dev.httpmarco.polocloud.agent.runtime.local.terminal.commands.CommandArgument
import dev.httpmarco.polocloud.agent.runtime.local.terminal.commands.CommandContext

class KeywordArgument(key: String) : CommandArgument<String>(key) {
    override fun defaultArgs(context: CommandContext): MutableList<String> {
        return listOf(key) as MutableList<String>
    }

    override fun wrongReason(rawInput: String): String {
        return ""
    }

    override fun buildResult(input: String, context: CommandContext): String {
        return ""
    }

    override fun predication(rawInput: String): Boolean {
        return super.predication(rawInput) && rawInput.equals(key, ignoreCase = true)
    }
}